src/objects/GameObject.cpp

```diff
@@ -6,9 +6,10 @@
 #include "core/API.h"
 
 GameObject::GameObject(sc2::Tag tag_): m_tag(tag_) {
+    m_unit = gAPI->observer().GetUnit(tag_);
```

Comment:
Do we really need this constructor now? Also, with this approach `ToUnit` should be removed.